environments/icds-new/postgresql.yml

```diff
@@ -1,66 +1,64 @@
 DEFAULT_POSTGRESQL_HOST: pgmain
-postgresql_dbs:
-  - django_alias: default
-    name: "{{commcarehq_main_db_name}}"
+USE_PARTITIONED_DATABASE: True
+REPORTING_DATABASES:
+  ucr: ucr
+  icds-ucr:
+    WRITE: icds-ucr
+    READ:
+      - [icds-ucr, 5]
+      - [icds-ucr-standby1, 5]
+  icds-test-ucr: icds-ucr
+
+dbs:
+  main:
     host: pgmain
-  - django_alias: synclogs
-    name: commcarehq_synclogs
-    host: pgsynclog
-  - django_alias: proxy
-    name: commcarehq_proxy
-    host: plproxy1
-  - django_alias: p1
-    name: commcarehq_p1
-    shards: [0, 103]
-    host: pgshard5
-  - django_alias: p2
-    name: commcarehq_p2
-    shards: [104, 205]
-    host: pgshard5
-  - django_alias: p3
-    name: commcarehq_p3
-    shards: [206, 308]
-    host: pgshard2
-  - django_alias: p4
-    name: commcarehq_p4
-    shards: [309, 411]
-    host: pgshard2
-  - django_alias: p5
-    name: commcarehq_p5
-    shards: [412, 514]
-    host: pgshard2
-  - django_alias: p6
-    name: commcarehq_p6
-    shards: [515, 617]
-    host: pgshard1
-  - django_alias: p7
-    name: commcarehq_p7
-    shards: [618, 720]
-    host: pgshard1
-  - django_alias: p8
-    name: commcarehq_p8
-    shards: [721, 823]
-    host: pgshard1
-  - django_alias: p9
-    name: commcarehq_p9
-    shards: [824, 926]
-    host: pgshard3
-  - django_alias: p10
-    name: commcarehq_p10
-    shards: [927, 1023]
-    host: pgshard3
-  - name: commcarehq_ucr
-    host: pgucr
-    django_alias: ucr
-    django_migrate: False
-  - django_alias: icds-ucr
-    name: commcarehq_icds_aggregatedata
+  formplayer:
     host: pgucr
-    query_stats: True
-  - name: "{{ formplayer_db_name }}"
+  ucr:
     host: pgucr
-  - django_alias: icds-ucr-standby1
-    name: commcarehq_icds_aggregatedata
-    host: pgucrstandby
-    create: False
-    django_migrate: False
+  synclogs:
+    host: pgsynclog
+  form_processing:
+    proxy:
+      host: plproxy1
+    partitions:
+      p1:
+        shards: [0, 103]
+        host: pgshard5
+      p2:
+        shards: [104, 205]
+        host: pgshard5
+      p3:
+        shards: [206, 308]
+        host: pgshard2
+      p4:
+        shards: [309, 411]
+        host: pgshard2
+      p5:
+        shards: [412, 514]
+        host: pgshard2
+      p6:
+        shards: [515, 617]
+        host: pgshard1
+      p7:
+        shards: [618, 720]
+        host: pgshard1
+      p8:
+        shards: [721, 823]
+        host: pgshard1
+      p9:
+        shards: [824, 926]
+        host: pgshard3
+      p10:
+        shards: [927, 1023]
+        host: pgshard3
+  custom:
+    - django_alias: icds-ucr
+      name: commcarehq_icds_aggregatedata
+      host: pgucr
+      query_stats: True
```

Comment:
Perhaps a note should be added about `django_migrate: True` here? It's an uncommon scenario, but necessary for this.

Also maybe custom should always have `django_migrate` default to `False` as that's the most common

Reply:
Good idea about leaving a comment and maybe an explicit `django_migrate: True` here. I'm a bit torn about giving `custom` it's own different defaults. While I agree that both in current usage and in future usage I can think of `django_migrate` is more likely to be `False` than `True`, I think it's also confusing to have a different default for `custom`, which is meant to be the sort of most manual/explicit way to add databases when they don't fit into predetermined categories. I think in that spirit, I'm going to make `django_migrate` required but have no default, meaning you have to be explicit about it for every custom db. The thinking there is that since it's custom, there's no inference we can reliably make about it, and either assumption could create a bit of a mess, so we'll just force you to think about it for each db. That work? Thanks for the suggestion.

Reply:
Sounds like a good compromise to me. Sounds like this leaves less room for error